mmv1/products/discoveryengine/ChatEngine.yaml

```diff
@@ -0,0 +1,163 @@
+  ]
+
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/discovery_engine_chat_engine_hardcode_solution_type.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engineId'
+    description: |
+      The ID to use for chat engine.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the chat engine. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/engines/{engine_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
+    description: |
+      The industry vertical that the chat engine registers.
+    values:
+      - :GENERIC
+    immutable: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      displayName
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    min_size: 1
+    description: |
+      Data Store IDs, currently modification of Data Stores after creation is not supported.
+    required: true
+    item_type: Api::Type::String
+    immutable: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'chatEngineConfig'
+    description: |
+      Configurations for a chat Engine.
+    required: true
+    immutable: true
+    ignore_read: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'agentCreationConfig'
+        description: |
+          Configurations for a chat Engine.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'business'
+            description: |
+              Business Name
+          - !ruby/object:Api::Type::String
+            name: 'defaultLanguageCode'
+            description: |
+              Default Language Code
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'timeZone'
+            description: |
+              time zone
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'location'
+            description: |
+              location
```

Comment:
```suggestion
              Agent location for Agent creation, supported values: global/us/eu. If not provided, us Engine will create Agent using us-central-1 by default; eu Engine will create Agent using eu-west-1 by default.
```

Reply:
normally this would require some kind of diff suppression or default-from-api behavior but since the fields are ignore_read anyway it shouldn't matter.